dist/resources/orders.js

```diff
@@ -64,11 +64,12 @@ module.exports = function (api) {
           receipt = params.receipt,
           payment_capture = params.payment_capture,
           notes = params.notes,
-          otherParams = _objectWithoutProperties(params, ['amount', 'currency', 'receipt', 'payment_capture', 'notes']);
+          method = params.method,
+          otherParams = _objectWithoutProperties(params, ['amount', 'currency', 'receipt', 'payment_capture', 'notes', 'method']);
 
       currency = currency || 'INR';
 
-      if (!amount) {
+      if (!amount && method !== 'emandate') {
```

Comment:
We should really check for amount being exactly zero. This could match for amount being absent as well.